### PR TITLE
distributed: added upgrade guide, updated templates

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
added upgrade guide and changelog for a future distributed chart release
updated common chart dependency
added `.Values.zoneTpl` for default config for each availability zone

### CODE REVIEW

File: charts/victoria-metrics-distributed/values.yaml [hunk 1]
```diff
@@ -33,229 +38,131 @@ write:
...
-# and write data to above vmauth-global-ingest endpoint.
-extraVMAgent:
-  enabled: true
-  name: test-vmagent
-  spec:
-    selectAllByDefault: true
+# and write data to above write-global endpoint.
+extra:
+  vmagent:
+    enabled: false
```
Comment (Haleygo) [inline charts/victoria-metrics-distributed/values.yaml:162]: I’d like to keep it enabled by default to write data to the write-global endpoint, allowing users to see data the first time they try this chart, and they can always disable it later.

File: charts/victoria-metrics-distributed/values.yaml [hunk 2]
```diff
@@ -33,229 +38,131 @@ write:
...
+# -- Defines a template for each availability zone
+zoneTpl:
+  common:
+    # -- Common for [VMAgent](https://docs.victoriametrics.com/operator/api/#vmagentspec),
+    # [VMAuth](https://docs.victoriametrics.com/operator/api/#vmauthspec),
+    # [VMCluster](https://docs.victoriametrics.com/operator/api/#vmclusterspec) spec params, like nodeSelector,
+    # affinity, topologySpreadConstraint, etc
+    spec:
+      nodeSelector:
+        topology.kubernetes.io/zone: '{{ (.zone).name }}'
```
[the remark this answers is not in the document]
Comment (AndrewChubatiuk) [reply charts/victoria-metrics-distributed/values.yaml:64]: this a `zoneTpl`, which can be redefined for each availability zone independently. also zoneTpl can be redefined

File: charts/victoria-metrics-distributed/README.md [hunk 3]
```diff
@@ -22,17 +22,17 @@ The default setup is as shown below:
 
 For write:
 1. extra-vmagent(optional): scrapes external targets and all the components installed by this chart, sends data to global write entrypoint.
-2. vmauth-global-write: global write entrypoint, proxies requests to one of the zone `vmagent` with `least_loaded` policy.
```
[the remark this answers is not in the document]
Comment (AndrewChubatiuk) [reply charts/victoria-metrics-distributed/README.md:25]: updated it

File: charts/victoria-metrics-distributed/values.yaml (hunk 1 above)
Comment (AndrewChubatiuk) [reply charts/victoria-metrics-distributed/values.yaml]: updated

File: charts/victoria-metrics-distributed/values.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Haleygo) [reply charts/victoria-metrics-distributed/values.yaml:64]: Could you add some example comments? For instance, having different resources setting for different zone vmcluster.

File: charts/victoria-metrics-distributed/values.yaml [hunk 4]
```diff
@@ -33,229 +38,131 @@ write:
...
+      vmselect:
+        replicaCount: 2
+        extraArgs: {}
+        resources: {}
+      vminsert:
+        replicaCount: 2
+        extraArgs: {}
+        resources: {}
+
+# -- Config per availability zone components, including vmagent, vmcluster, vmauth etc
```
[the remark this answers is not in the document]
Comment (AndrewChubatiuk) [reply charts/victoria-metrics-distributed/values.yaml]: updated

File: charts/victoria-metrics-distributed/values.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (AndrewChubatiuk) [reply charts/victoria-metrics-distributed/values.yaml:65]: added examples

**Review by Haleygo — APPROVED** [review]
LGTM!

**Pull request merged**